deep_qa/data/data_generator.py

```diff
@@ -46,6 +46,18 @@ class DataGenerator:
         certain padding values or not.  It's handled correctly for ``num_sentence_words`` and
         ``num_word_characters`` in :class:`~deep_qa.training.TextTrainer`, but you need to be sure
         to implement it correctly in subclasses for this to work.
+    permute_probability: double, optional (default=.2)
```

Comment:
I'm not sure that this is really the best way to do this - imagine a dataset with binary labels where all sentences of length > 10 are positive and all sentences of length < 10 are negative (obviously an exaggerated case but the point holds). Using this method of shuffling only one instance with a negative label will cross over from being in a batch with all negative labels to being in a batch with all positive labels and even then this will only happen with probability `permute_probability`. 

What about doing something based off making the distribution of labels per batch more uniform past some threshold, say 60-70%? This is a hard problem to tackle.